packages/core/src/components/TextField.tsx

```diff
@@ -398,23 +398,56 @@ class TextField extends React.Component<Props> {
       type === "solid" ? { marginHorizontal: 12 } : {},
     ];
 
+    const {
+      backgroundColor: bgColor,
+      padding,
+      paddingTop,
+      paddingBottom,
+      paddingLeft,
+      paddingRight,
+      borderRadius,
+      borderWidth,
+      borderColor: borderCol,
+      ...styleProp
+    } = StyleSheet.flatten(style || {}) as ViewStyle & { height?: number };
+
     return (
-      <View style={[styles.container, style]}>
+      <View style={[styles.container, styleProp]}>
         {leftIconName && leftIconMode === "outset" ? (
           <Icon {...leftIconProps} style={leftIconStyle} />
         ) : null}
-        <View style={[containerStyle, style ? { height: style.height } : {}]}>
+        <View
+          style={[
+            containerStyle,
+            style
+              ? {
+                  ...(style.height ? { height: style.height } : {}),
```

Comment:
have you tried

```
height: style.height,
height: style.height ?? undefined,
height: style.height || undefined
height: style.height ? style.height : undefined // this one for sure works
```

Reply:
Yeah - surprisingly, those will not work. See this Snack - `undefined` overrides an explicitly set value.

https://snack.expo.io/@donald/groaning-edamame